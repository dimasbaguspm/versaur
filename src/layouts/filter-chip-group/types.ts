import type { HTMLAttributes, ReactNode } from 'react'
import type { VariantProps } from '@/utils/variants'
import type { filterChipGroupVariants } from './helpers'

/**
 * Props for the FilterChipGroup component
 */
export interface FilterChipGroupProps
  extends HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof filterChipGroupVariants> {
  /**
   * Children (should be FilterChip components)
   */
  children: ReactNode

  /**
   * Alignment of filter chips within the group
   * @default 'start'
   */
  alignment?: 'start' | 'center' | 'end' | 'between' | 'around' | 'evenly'

  /**
   * Whether filter chips should be fluid (full width)
   * @default false
   */
  fluid?: boolean

  /**
   * Orientation of the filter chip group
   * @default 'horizontal'
   */
  orientation?: 'horizontal' | 'vertical'

  /**
   * Size of the gap between filter chips
   * @default 'md'
   */
  gap?: 'xs' | 'sm' | 'md' | 'lg' | 'xl'

  /**
   * Whether the badge group has bottom margin
   * @default false
   */
  hasMargin?: boolean

  /**
   * Whether items should overlay in single line without wrapping
   * @default false
   */
  overlay?: boolean
}
