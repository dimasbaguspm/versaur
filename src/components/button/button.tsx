import React from 'react'
import { cn } from '@/utils/cn'
import { cva } from '@/utils/variants'
import type { ButtonProps } from './types'

const buttonVariants = cva(
  'inline-flex items-center justify-center gap-2 rounded-md font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none active:scale-[0.98] select-none',
  {
    variants: {
      variant: {
        // Core variants using Versaur color system
        primary:
          'bg-coral text-white hover:bg-coral/90 focus-visible:ring-coral-soft focus-visible:ring-offset-white shadow-sm hover:shadow-md',
        secondary:
          'bg-sage text-white hover:bg-sage/90 focus-visible:ring-sage-soft focus-visible:ring-offset-white shadow-sm hover:shadow-md',
        tertiary:
          'bg-mist text-white hover:bg-mist/90 focus-visible:ring-mist-soft focus-visible:ring-offset-white shadow-sm hover:shadow-md',
        ghost:
          'bg-white text-foreground hover:bg-slate-soft focus-visible:ring-slate-soft focus-visible:ring-offset-white',
        neutral:
          'bg-neutral text-foreground border border-border hover:bg-neutral/80 focus-visible:ring-foreground-soft focus-visible:ring-offset-white shadow-sm',

        // Outline variants
        'primary-outline':
          'border border-coral text-coral bg-white hover:bg-coral hover:text-white focus-visible:ring-coral-soft focus-visible:ring-offset-white transition-all',
        'secondary-outline':
          'border border-sage text-sage bg-white hover:bg-sage hover:text-white focus-visible:ring-sage-soft focus-visible:ring-offset-white transition-all',
        'tertiary-outline':
          'border border-mist text-mist bg-white hover:bg-mist hover:text-white focus-visible:ring-mist-soft focus-visible:ring-offset-white transition-all',
        'ghost-outline':
          'border border-slate text-slate bg-white hover:bg-slate hover:text-white focus-visible:ring-slate-soft focus-visible:ring-offset-white transition-all',
        'neutral-outline':
          'border border-neutral text-foreground bg-white hover:bg-neutral hover:text-foreground focus-visible:ring-foreground-soft focus-visible:ring-offset-white transition-all',

        // Ghost variants (subtle)
        'primary-ghost':
          'text-coral bg-white hover:bg-coral-soft focus-visible:ring-coral-soft focus-visible:ring-offset-white',
        'secondary-ghost':
          'text-sage bg-white hover:bg-sage-soft focus-visible:ring-sage-soft focus-visible:ring-offset-white',
        'tertiary-ghost':
          'text-mist bg-white hover:bg-mist-soft focus-visible:ring-mist-soft focus-visible:ring-offset-white',
        'neutral-ghost':
          'text-foreground bg-white hover:bg-neutral/70 focus-visible:ring-foreground-soft focus-visible:ring-offset-white',

        // Semantic variants
        success:
          'bg-success text-white hover:bg-success/90 focus-visible:ring-success-soft focus-visible:ring-offset-white shadow-sm hover:shadow-md',
        'success-outline':
          'border border-success text-success bg-white hover:bg-success hover:text-white focus-visible:ring-success-soft focus-visible:ring-offset-white transition-all',
        'success-ghost':
          'text-success bg-white hover:bg-success-soft focus-visible:ring-success-soft focus-visible:ring-offset-white',

        info: 'bg-info text-white hover:bg-info/90 focus-visible:ring-info-soft focus-visible:ring-offset-white shadow-sm hover:shadow-md',
        'info-outline':
          'border border-info text-info bg-white hover:bg-info hover:text-white focus-visible:ring-info-soft focus-visible:ring-offset-white transition-all',
        'info-ghost':
          'text-info bg-white hover:bg-info-soft focus-visible:ring-info-soft focus-visible:ring-offset-white',

        warning:
          'bg-warning text-white hover:bg-warning/90 focus-visible:ring-warning-soft focus-visible:ring-offset-white shadow-sm hover:shadow-md',
        'warning-outline':
          'border border-warning text-warning bg-white hover:bg-warning hover:text-white focus-visible:ring-warning-soft focus-visible:ring-offset-white transition-all',
        'warning-ghost':
          'text-warning bg-white hover:bg-warning-soft focus-visible:ring-warning-soft focus-visible:ring-offset-white',

        danger:
          'bg-danger text-white hover:bg-danger/90 focus-visible:ring-danger-soft focus-visible:ring-offset-white shadow-sm hover:shadow-md',
        'danger-outline':
          'border border-danger text-danger bg-white hover:bg-danger hover:text-white focus-visible:ring-danger-soft focus-visible:ring-offset-white transition-all',
        'danger-ghost':
          'text-danger bg-white hover:bg-danger-soft focus-visible:ring-danger-soft focus-visible:ring-offset-white',

        // Utility variants
        outline:
          'border border-border text-foreground bg-white hover:bg-accent-soft focus-visible:ring-accent-soft focus-visible:ring-offset-white transition-all',
        destructive:
          'bg-danger text-white hover:bg-danger/90 focus-visible:ring-danger-soft focus-visible:ring-offset-white shadow-sm hover:shadow-md',
      },
      size: {
        sm: 'h-8 px-3 text-sm min-w-[2.25rem]',
        md: 'h-10 px-4 text-sm min-w-[2.5rem]',
        lg: 'h-12 px-8 text-lg min-w-[2.75rem]',
      },
    },
    defaultVariants: {
      variant: 'primary',
      size: 'md',
    },
  }
)

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  function Button(
    {
      className,
      variant = 'primary',
      size = 'md',
      disabled = false,
      type = 'button',
      children,
      ...props
    },
    ref
  ) {
    return (
      <button
        ref={ref}
        type={type}
        className={cn(buttonVariants({ variant, size }), className)}
        disabled={disabled}
        aria-disabled={disabled}
        inert={disabled ? true : undefined}
        {...props}
      >
        {children}
      </button>
    )
  }
)
