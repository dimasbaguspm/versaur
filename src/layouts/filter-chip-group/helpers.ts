import { cva } from '@/utils/variants'

/**
 * Filter chip group variants for managing layout and alignment
 */
export const filterChipGroupVariants = cva('flex flex-wrap items-center', {
  variants: {
    orientation: {
      horizontal: 'flex-row',
      vertical: 'flex-col',
    },
    alignment: {
      start: 'justify-start',
      center: 'justify-center',
      end: 'justify-end',
      between: 'justify-between',
      around: 'justify-around',
      evenly: 'justify-evenly',
    },
    gap: {
      xs: 'gap-1',
      sm: 'gap-2',
      md: 'gap-3',
      lg: 'gap-4',
      xl: 'gap-6',
    },
    fluid: {
      true: '[&>*]:flex-1',
      false: '',
    },
  },
  defaultVariants: {
    orientation: 'horizontal',
    alignment: 'start',
    gap: 'md',
    fluid: false,
  },
})
