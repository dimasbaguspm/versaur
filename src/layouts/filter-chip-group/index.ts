export { FilterChipGroup } from './filter-chip-group'
export type { FilterChipGroupProps } from './types'
