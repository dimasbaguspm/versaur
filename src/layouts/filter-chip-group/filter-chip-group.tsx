import { cn } from '@/utils/cn'
import { filterChipGroupVariants } from './helpers'
import type { FilterChipGroupProps } from './types'
import { forwardRef } from 'react'

/**
 * FilterChipGroup provides layout management for filter chips with control over
 * positioning, alignment, and fluid behavior
 *
 * @example
 * <FilterChipGroup alignment="center" gap="sm">
 *   <FilterChip active>Category 1</FilterChip>
 *   <FilterChip>Category 2</FilterChip>
 * </FilterChipGroup>
 *
 * @example
 * // Vertical filter chip group
 * <FilterChipGroup orientation="vertical" fluid>
 *   <FilterChip active>Category 1</FilterChip>
 *   <FilterChip>Category 2</FilterChip>
 *   <FilterChip>Category 3</FilterChip>
 * </FilterChipGroup>
 *
 * @example
 * // Fluid filter chips with space between
 * <FilterChipGroup alignment="between" fluid>
 *   <FilterChip>Clear All</FilterChip>
 *   <FilterChip active>Apply</FilterChip>
 * </FilterChipGroup>
 */
export const FilterChipGroup = forwardRef<HTMLDivElement, FilterChipGroupProps>(
  function FilterChipGroup(
    {
      className,
      children,
      orientation = 'horizontal',
      alignment = 'start',
      gap = 'md',
      fluid = false,
      ...props
    },
    ref
  ) {
    return (
      <div
        ref={ref}
        className={cn(
          filterChipGroupVariants({
            orientation,
            alignment,
            gap,
            fluid,
          }),
          className
        )}
        role='group'
        {...props}
      >
        {children}
      </div>
    )
  }
)
