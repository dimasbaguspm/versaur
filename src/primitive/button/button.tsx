import React from 'react'
import { cn } from '@/utils/cn'
import type { ButtonProps } from './types'
import { buttonVariants } from './helpers'
import { Icon } from '../icon'
import { LoaderCircleIcon } from 'lucide-react'

/**
 * Button component following browser-standard button behavior
 * Supports 4 variants: primary, ghost, outline, and destructive
 * Provides 3 sizes: sm, md (default), lg
 */
export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  function Button(
    {
      className,
      variant = 'primary',
      size = 'md',
      disabled = false,
      type = 'button',
      busy,
      children,
      ...props
    },
    ref
  ) {
    // Map button sizes to icon sizes
    const iconSize = size === 'sm' ? 'sm' : size === 'lg' ? 'md' : 'sm'

    // Determine icon color based on variant
    const getIconColor = () => {
      if (variant === 'primary' || variant === 'destructive') {
        return 'white'
      }
      if (variant === 'outline' || variant === 'ghost') {
        return 'inherit'
      }
      return 'inherit'
    }

    return (
      <button
        ref={ref}
        type={type}
        className={cn(buttonVariants({ variant, size }), className)}
        disabled={disabled || busy}
        aria-disabled={disabled || busy}
        inert={disabled || busy ? true : undefined}
        aria-busy={busy ? true : undefined}
        {...props}
      >
        {busy && (
          <Icon
            as={LoaderCircleIcon}
            size={iconSize}
            color={getIconColor()}
            className='animate-spin'
          />
        )}
        {children}
      </button>
    )
  }
)
