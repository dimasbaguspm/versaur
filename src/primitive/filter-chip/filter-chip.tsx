import React from 'react'
import { cn } from '@/utils/cn'
import type { FilterChipProps } from './types'
import { filterChipVariants } from './helpers'
import { Icon } from '../icon'
import { XIcon } from 'lucide-react'

export const FilterChip = React.forwardRef<HTMLButtonElement, FilterChipProps>(
  function FilterChip(
    { className, size = 'md', disabled = false, children, ...props },
    ref
  ) {
    return (
      <button
        ref={ref}
        type='button'
        className={cn(filterChipVariants({ size }), className)}
        disabled={disabled}
        aria-disabled={disabled}
        aria-label={`${children}, removable`}
        inert={disabled ? true : undefined}
        {...props}
      >
        <span className='flex-shrink-0'>{children}</span>

        <Icon as={XIcon} color='inherit' size={size === 'lg' ? 'md' : 'sm'} />
      </button>
    )
  }
)
