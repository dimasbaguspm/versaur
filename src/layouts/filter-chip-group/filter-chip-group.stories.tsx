import type { Meta, StoryObj } from '@storybook/react'
import { FilterChipGroup } from './filter-chip-group'
import { FilterChip } from '@/primitive/filter-chip'
import { Button } from '@/primitive'

/**
 * FilterChipGroup provides layout management for filter chips with control over positioning,
 * alignment, and fluid behavior. It's designed to wrap FilterChip components and manage
 * their positioning effectively for filtering interfaces.
 *
 * ## Features
 * - **Alignment control**: Position filter chips with start, center, end, between, around, or evenly
 * - **Orientation**: Horizontal or vertical layout
 * - **Gap management**: Control spacing between filter chips with predefined sizes
 * - **Overlay mode**: Single-line horizontal scrolling without wrapping
 * - **Accessibility**: Proper ARIA roles for screen readers
 *
 * ## Usage
 * Wrap FilterChip components with FilterChipGroup to control their layout and alignment
 */
const meta: Meta<typeof FilterChipGroup> = {
  title: 'Layouts/FilterChipGroup',
  component: FilterChipGroup,
  parameters: {
    docs: {
      description: {
        component:
          'A flexible layout component for managing filter chip positioning, alignment, and fluid behavior.',
      },
    },
  },
  argTypes: {
    alignment: {
      control: { type: 'select' },
      options: ['start', 'center', 'end', 'between', 'around', 'evenly'],
      description: 'Alignment of filter chips within the group',
    },
    orientation: {
      control: { type: 'select' },
      options: ['horizontal', 'vertical'],
      description: 'Layout direction of the filter chip group',
    },
    gap: {
      control: { type: 'select' },
      options: ['xs', 'sm', 'md', 'lg', 'xl'],
      description: 'Size of the gap between filter chips',
    },
    fluid: {
      control: { type: 'boolean' },
      description: 'Whether filter chips should expand to fill available space',
    },
    overlay: {
      control: { type: 'boolean' },
      description:
        'Whether items should overlay in single line without wrapping',
    },
  },
  tags: ['autodocs'],
}

export default meta
type Story = StoryObj<typeof FilterChipGroup>

/**
 * Default filter chip group with horizontal layout and start alignment
 */
export const Default: Story = {
  render: args => (
    <FilterChipGroup {...args}>
      <FilterChip>Category: Electronics</FilterChip>
      <FilterChip>Brand: Samsung</FilterChip>
      <FilterChip>Price: $500-$1000</FilterChip>
      <Button variant='ghost' size='sm'>
        Clear All
      </Button>
    </FilterChipGroup>
  ),
}

/**
 * Filter chip group with space between - useful for filter sections with clear/apply actions
 */
export const SpaceBetween: Story = {
  render: args => (
    <div
      style={{
        width: '500px',
        border: '1px solid var(--color-border)',
        padding: '16px',
      }}
    >
      <FilterChipGroup {...args} alignment='between'>
        <div className='flex gap-2 flex-wrap'>
          <FilterChip>Category: Electronics</FilterChip>
          <FilterChip>Brand: Samsung</FilterChip>
        </div>
        <Button variant='ghost' size='sm'>
          Clear All
        </Button>
      </FilterChipGroup>
    </div>
  ),
}

/**
 * Vertical filter chip group - useful for sidebars or stacked filter interfaces
 */
export const Vertical: Story = {
  render: args => (
    <div
      style={{
        width: '250px',
        border: '1px solid var(--color-border)',
        padding: '16px',
      }}
    >
      <FilterChipGroup {...args} orientation='vertical' gap='sm'>
        <FilterChip>Category: Electronics</FilterChip>
        <FilterChip>Brand: Samsung</FilterChip>
        <FilterChip>Price: $500-$1000</FilterChip>
        <Button variant='ghost' size='sm'>
          Clear Filters
        </Button>
      </FilterChipGroup>
    </div>
  ),
}

/**
 * Overlay mode - filter chips scroll horizontally without wrapping
 */
export const Overlay: Story = {
  render: args => (
    <div
      style={{
        width: '400px',
        border: '1px solid var(--color-border)',
        padding: '16px',
      }}
    >
      <FilterChipGroup {...args} overlay>
        <FilterChip>Category: Electronics</FilterChip>
        <FilterChip>Brand: Samsung</FilterChip>
        <FilterChip>Price: $500-$1000</FilterChip>
        <FilterChip>Rating: 4+</FilterChip>
        <FilterChip>In Stock</FilterChip>
        <Button variant='ghost' size='sm'>
          Clear All
        </Button>
      </FilterChipGroup>
    </div>
  ),
}

/**
 * Real-world e-commerce filter example
 */
export const RealWorldExamples: Story = {
  render: _args => (
    <div className='space-y-6 max-w-2xl'>
      {/* Active filters header */}
      <div>
        <h3 className='mb-3 text-sm font-medium'>Active Filters (8)</h3>
        <FilterChipGroup alignment='start' gap='sm'>
          <FilterChip>Category: Electronics</FilterChip>
          <FilterChip>Brand: Samsung</FilterChip>
          <FilterChip>Price: $500-$1000</FilterChip>
          <FilterChip>Rating: 4+</FilterChip>
          <FilterChip>Color: Black</FilterChip>
          <FilterChip>In Stock</FilterChip>
          <FilterChip>Free Shipping</FilterChip>
          <FilterChip>On Sale</FilterChip>
          <Button variant='ghost' size='sm'>
            Clear All
          </Button>
        </FilterChipGroup>
      </div>

      {/* Sidebar filters */}
      <div
        style={{
          width: '250px',
          border: '1px solid var(--color-border)',
          padding: '16px',
        }}
      >
        <h3 className='mb-3 text-sm font-medium'>Sidebar Filters</h3>
        <FilterChipGroup orientation='vertical' gap='sm'>
          <FilterChip>Category: Electronics</FilterChip>
          <FilterChip>Price: $500-$1000</FilterChip>
          <FilterChip>In Stock</FilterChip>
        </FilterChipGroup>
      </div>
    </div>
  ),
}
