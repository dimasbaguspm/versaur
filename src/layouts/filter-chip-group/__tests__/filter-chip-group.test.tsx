import { describe, it, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import { composeStories } from '@storybook/react'
import * as stories from '../filter-chip-group.stories'
import { FilterChipGroup } from '../filter-chip-group'
import { FilterChip } from '@/primitive/filter-chip'
import { Button } from '@/primitive/button'

const composedStories = composeStories(stories)

describe('FilterChipGroup', () => {
  it('should render HTML correctly', () => {
    const { asFragment } = render(
      <FilterChipGroup>
        <FilterChip>Active Filter 1</FilterChip>
        <FilterChip>Active Filter 2</FilterChip>
        <Button variant='ghost'>Clear All</Button>
      </FilterChipGroup>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render with role group and default classes', () => {
    render(
      <FilterChipGroup data-testid='filter-chip-group'>
        <FilterChip>Filter 1</FilterChip>
        <FilterChip>Filter 2</FilterChip>
      </FilterChipGroup>
    )

    const filterChipGroup = screen.getByTestId('filter-chip-group')
    expect(filterChipGroup).toBeInTheDocument()
    expect(filterChipGroup).toHaveAttribute('role', 'group')
    expect(filterChipGroup).toHaveClass('flex', 'flex-wrap', 'items-center')
  })

  it('should apply orientation variants', () => {
    const { rerender } = render(
      <FilterChipGroup orientation='horizontal' data-testid='filter-chip-group'>
        <FilterChip>Filter 1</FilterChip>
      </FilterChipGroup>
    )

    let filterChipGroup = screen.getByTestId('filter-chip-group')
    expect(filterChipGroup).toHaveClass('flex-row')

    rerender(
      <FilterChipGroup orientation='vertical' data-testid='filter-chip-group'>
        <FilterChip>Filter 1</FilterChip>
      </FilterChipGroup>
    )

    filterChipGroup = screen.getByTestId('filter-chip-group')
    expect(filterChipGroup).toHaveClass('flex-col')
  })

  it('should apply overlay mode', () => {
    render(
      <FilterChipGroup overlay data-testid='filter-chip-group'>
        <FilterChip>Filter 1</FilterChip>
        <FilterChip>Filter 2</FilterChip>
      </FilterChipGroup>
    )

    const filterChipGroup = screen.getByTestId('filter-chip-group')
    expect(filterChipGroup).toHaveClass('flex-nowrap', 'overflow-x-auto')
  })

  it('should apply fluid behavior', () => {
    render(
      <FilterChipGroup fluid data-testid='filter-chip-group'>
        <FilterChip>Filter 1</FilterChip>
      </FilterChipGroup>
    )

    const filterChipGroup = screen.getByTestId('filter-chip-group')
    expect(filterChipGroup).toHaveClass('[&>*]:flex-1')
  })

  it('should forward ref correctly', () => {
    const ref = { current: null }
    render(
      <FilterChipGroup ref={ref}>
        <FilterChip>Filter 1</FilterChip>
      </FilterChipGroup>
    )

    expect(ref.current).toBeInstanceOf(HTMLDivElement)
  })

  it('should accept custom className', () => {
    render(
      <FilterChipGroup className='custom-class' data-testid='filter-chip-group'>
        <FilterChip>Filter 1</FilterChip>
      </FilterChipGroup>
    )

    const filterChipGroup = screen.getByTestId('filter-chip-group')
    expect(filterChipGroup).toHaveClass('custom-class')
  })

  describe('Stories', () => {
    Object.entries(composedStories).forEach(([name, Story]) => {
      it(`should render ${name} story`, () => {
        const { container } = render(<Story />)
        expect(container.firstChild).toBeInTheDocument()
      })
    })
  })
})
