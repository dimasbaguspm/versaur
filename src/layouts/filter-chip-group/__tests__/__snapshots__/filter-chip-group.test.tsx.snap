// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterChipGroup > should render HTML correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-wrap items-center flex-row justify-start gap-3"
    role="group"
  >
    <button
      aria-disabled="false"
      aria-label="Active Filter 1, removable"
      class="inline-flex items-center gap-1.5 rounded-full font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none select-none cursor-pointer border border-border text-foreground bg-white hover:bg-neutral focus-visible:ring-foreground-light focus-visible:ring-offset-white h-8 px-4 text-sm"
      type="button"
    >
      <span
        class="flex-shrink-0"
      >
        Active Filter 1
      </span>
      <svg
        aria-hidden="true"
        class="lucide lucide-x inline-flex items-center justify-center w-4 h-4"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18 6 6 18"
        />
        <path
          d="m6 6 12 12"
        />
      </svg>
    </button>
    <button
      aria-disabled="false"
      aria-label="Active Filter 2, removable"
      class="inline-flex items-center gap-1.5 rounded-full font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none select-none cursor-pointer border border-border text-foreground bg-white hover:bg-neutral focus-visible:ring-foreground-light focus-visible:ring-offset-white h-8 px-4 text-sm"
      type="button"
    >
      <span
        class="flex-shrink-0"
      >
        Active Filter 2
      </span>
      <svg
        aria-hidden="true"
        class="lucide lucide-x inline-flex items-center justify-center w-4 h-4"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18 6 6 18"
        />
        <path
          d="m6 6 12 12"
        />
      </svg>
    </button>
    <button
      aria-disabled="false"
      class="inline-flex items-center justify-center gap-2 rounded-md font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none active:scale-[0.98] select-none cursor-pointer text-foreground bg-white hover:bg-neutral/50 focus-visible:ring-foreground focus-visible:ring-offset-white h-9 px-4 text-sm min-w-[2.5rem]"
      type="button"
    >
      Clear All
    </button>
  </div>
</DocumentFragment>
`;
